src/shared/tests.c

```diff
@@ -42,5 +42,9 @@ const char* get_exe_relative_testdata_dir(void) {
 
         assert_se(readlink_and_make_absolute("/proc/self/exe", &exedir) >= 0);
         assert_se(snprintf(testdir, sizeof(testdir), "%s/testdata", dirname(exedir)) > 0);
+        if (access(testdir, F_OK) < 0) {
+                fprintf(stderr,  "Test data directory '%s' does not exist, set $TEST_DIR\n", testdir);
+                exit(1);
```

Comment:
> and let the test fail by itself later?

our tests fail via `ABRT`. this prevents all cleanup operations too.

> Or just write it to stderr to get it as first error message

I think
```
$ ./libtool --mode=execute ./test-engine
Test data directory '/home/vagrant/systemd/.libs/testdata' does not exist, set $TEST_DIR
```
is better then
```
some stdout
...
Test data directory '/home/vagrant/systemd/.libs/testdata' does not exist, set $TEST_DIR
many lines
aborted
```